src/Data/Equality/Graph.hs

```diff
@@ -149,7 +153,7 @@ merge a b egr0 =
 
            -- Leader is the class with more parents
            (leader, leader_class, sub, sub_class) =
-               if (sizeNM (class_a^._parents)) < (sizeNM (class_b^._parents))
+               if sizeSL (class_a^._parents) < sizeSL (class_b^._parents)
```

Comment:
Ah, this is the reason for the cached size. Is this a typical criterion for choosing the head of the merge? I've never used that one. In Egg I would have imagined using the representative with the best grade (as specified by the user) as the canonical representative.

Reply:
:-) https://github.com/egraphs-good/egg/blob/e19a18f60a6f1d3f5c466b37c0fe4d3cd69e550e/src/egraph.rs#L738-L750

Reply:
Indeed.

Thinking back, since we constantly traverse this list for canonization/nubbing. It's ok to compute the length at the same time.

In fact, I realise as I'm writing this that the fact that the length is strict makes sure that the spine of the list, at least, is forced. It's probably a good thing.